packages/itwin/reports-config-widget/src/widget/components/BulkExtractor.ts

```diff
@@ -0,0 +1,188 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+import { ExtractionClient, REPORTING_BASE_PATH, ReportMapping, ReportsClient, MappingsClient, ExtractorState } from "@itwin/insights-client";
+import { generateUrl, handleError } from "./utils";
+import { useReportsApiConfig } from "../context/ReportsApiConfigContext";
+import type { CreateTypeFromInterface } from "./utils";
+import type {
+  GetSingleIModelParams,
+  IModelsClientOptions,
+} from "@itwin/imodels-client-management";
+import { Constants, IModelsClient } from "@itwin/imodels-client-management";
+import { AccessTokenAdapter } from "@itwin/imodels-access-frontend";
+import type { ReportsApiConfig } from "../context/ReportsApiConfigContext";
+import { ExtractionStates } from "./ExtractionStatus";
+
+export type ReportMappingType = CreateTypeFromInterface<ReportMapping>;
+
+export type ReportMappingAndMapping = ReportMappingType & {
+  mappingName: string;
+  mappingDescription: string;
+  iModelName: string;
+};
+
+export default class BulkExtractor {
+  reportStates = new Map<string, ExtractionStates>();
```

Comment:
Mark all fields as either `private` or `public`. I think these should be `private`. Then, all private fields should be prefixed with an underscore `_`.

Reply:
Prefixes for fields has been added